lib/auth/oidc.go

```diff
@@ -828,76 +685,9 @@ func (a *Server) getClaims(oidcClient *oidc.Client, connector types.OIDCConnecto
-	// issuer URL as Google Workspace (e.g.
-	// https://developers.google.com/identity/protocols/oauth2/openid-connect).
-	if connector.GetIssuerURL() == teleport.GSuiteIssuerURL && (connector.GetGoogleServiceAccountURI() != "" || connector.GetGoogleServiceAccount() != "") {
-		email, _, err := claims.StringClaim("email")
-		if err != nil {
-			return nil, trace.Wrap(err)
-		}
-
-		var jsonCredentials []byte
-		var credentialLoadingMethod string
-		if connector.GetGoogleServiceAccountURI() != "" {
-			// load the google service account from URI
-			credentialLoadingMethod = "google_service_account_uri"
-
-			uri, err := utils.ParseSessionsURI(connector.GetGoogleServiceAccountURI())
-			if err != nil {
-				return nil, trace.BadParameter("failed to parse google_service_account_uri: %v", err)
-			}
-			jsonCredentials, err = ioutil.ReadFile(uri.Path)
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-		} else if connector.GetGoogleServiceAccount() != "" {
-			// load the google service account from string
-			credentialLoadingMethod = "google_service_account"
-			jsonCredentials = []byte(connector.GetGoogleServiceAccount())
-		}
-
-		config, err := google.JWTConfigFromJSON(jsonCredentials, teleport.GSuiteGroupsScope)
-		if err != nil {
-			return nil, trace.BadParameter("unable to parse google service account from %v: %v", credentialLoadingMethod, err)
-		}
-
-		impersonateAdmin := connector.GetGoogleAdminEmail()
-		if impersonateAdmin == "" {
-			return nil, trace.NotFound(
-				"the google workspace connector requires google_admin_email user to impersonate, as service accounts can not be used directly https://developers.google.com/identity/protocols/OAuth2ServiceAccount#delegatingauthority")
-		}
-
-		// User should impersonate admin user, otherwise it won't work:
-		//
-		// https://developers.google.com/admin-sdk/directory/v1/guides/delegation
-		//
-		// "Note: Only users with access to the Admin APIs can access the Admin SDK Directory API, therefore your service account needs to impersonate one of those users to access the Admin SDK Directory API. Additionally, the user must have logged in at least once and accepted the G Suite Terms of Service."
-		//
-		domain, exists, err := userInfoClaims.StringClaim(teleport.GSuiteDomainClaim)
-		if err != nil || !exists {
-			return nil, trace.BadParameter("hd is the required claim for Google Workspace")
-		}
-		config.Subject = impersonateAdmin
-
-		gsuiteClaims, err := a.claimsFromGSuite(config, connector.GetIssuerURL(), email, domain)
-		if err != nil {
-			if !trace.IsNotFound(err) {
-				return nil, trace.Wrap(err)
-			}
-			log.Debugf("Found no Google Workspace claims.")
-		} else {
-			if gsuiteClaims != nil {
-				log.Debugf("Got gsuiteClaims claims from Google Workspace: %v.", gsuiteClaims)
-			}
-			claims, err = mergeClaims(claims, gsuiteClaims)
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-		}
+	claims, err = addGsuiteClaims(a.closeCtx, connector, claims)
```

Comment:
I think it's called Google Workspace now, so how about `addGoogleWorkspaceClaims`.